src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -998,5 +999,107 @@ static void RefParam(ref dynamic p)
+}
+public delegate ref TR RefFunc<T1, TR>(T1 t1);
+public delegate ref TR RefFunc<T1, T2, TR>(T1 t1, T2 t2);
+";
+            CreateCompilationWithMscorlibAndSystemCore(text)
+                .GetDiagnostics()
+                // It turns out each of them is diagnosed with ErrorCode.ERR_InvalidExprTerm in the midst
+                // of a flurry of other syntax errors.
+                .Where(d => d.Code == (int)ErrorCode.ERR_InvalidExprTerm)
+                .Verify(
+                // (9,39): error CS1525: Invalid expression term 'ref'
+                //         { var za = from x in a select ref x; } // error 1
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(9, 39),
+                // (10,42): error CS1525: Invalid expression term 'ref'
+                //         { var zc = from x in a from y in ref a select x; } // error2
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(10, 42),
+                // (11,46): error CS1525: Invalid expression term 'ref'
+                //         { var zd = from x in a from int y in ref a select x; } // error 3
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(11, 46),
+                // (12,42): error CS1525: Invalid expression term 'ref'
+                //         { var ze = from x in a from y in ref a where true select x; } // error 4
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(12, 42),
+                // (13,46): error CS1525: Invalid expression term 'ref'
+                //         { var zf = from x in a from int y in ref a where true select x; } // error 5
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(13, 46),
+                // (14,40): error CS1525: Invalid expression term 'ref'
+                //         { var zg = from x in a let y = ref a select x; } // error 6
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(14, 40),
+                // (15,38): error CS1525: Invalid expression term 'ref'
+                //         { var zh = from x in a where ref b select x; } // error 7
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(15, 38),
+                // (16,42): error CS1525: Invalid expression term 'ref'
+                //         { var zi = from x in a join y in ref a on x equals y select x; } // error 8 (not lambda case)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(16, 42),
+                // (16,42): error CS1525: Invalid expression term 'ref'
+                //         { var zi = from x in a join y in ref a on x equals y select x; } // error 8 (not lambda case)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(16, 42),
```

Comment:
Agreed, it is the same syntax error arising from a different place in the error recovery for this source.